src/interface/sparse_pack_base.cpp

```diff
@@ -169,23 +167,25 @@ SparsePackBase SparsePackBase::Build(T *pmd, const PackDescriptor &desc) {
         }
       }
 
-      pack.bounds_h_(1, b, i) = idx - 1;
-
-      if (pack.bounds_h_(1, b, i) < pack.bounds_h_(0, b, i)) {
-        // Did not find any allocated variables meeting our criteria
-        pack.bounds_h_(0, b, i) = -1;
-        // Make the upper bound more negative so a for loop won't iterate once
-        pack.bounds_h_(1, b, i) = -2;
+      if (!desc.flat) {
+        pack.bounds_h_(1, b, i) = idx - 1;
+        if (pack.bounds_h_(1, b, i) < pack.bounds_h_(0, b, i)) {
+          // Did not find any allocated variables meeting our criteria
+          pack.bounds_h_(0, b, i) = -1;
+          // Make the upper bound more negative so a for loop won't iterate once
+          pack.bounds_h_(1, b, i) = -2;
+        }
```

Comment:
Would have to change this back as well for the bounds to make sense.

Reply:
yes. Fixed.